src/providers/object-provider.js

```diff
@@ -138,8 +169,14 @@ export default class YamcsObjectProvider {
                 namespace: this.namespace
             },
             name: parameter.name
-        };
-        if (parameter.type.engType === 'aggregate') {
+        }
+
+        let isAggregate = false;
+        if ('type' in parameter && parameter.type.engType === 'aggregate') {
```

Comment:
There's the potential for a subtle bug here. Using `in` will return true if the key exists on the object, even if the corresponding value is `undefined`. This will cause a problem when the RHS of this statement is evaluated.

This also probably applies in some of the other places where `in` has been used here. An alternative is to use 
``` JavaScript
if(parameter.type === undefined && parameter.type.engType === 'aggregate')
```

Reply:
Will fix.